Source/WTF/wtf/ArgumentCoder.h

```diff
@@ -34,16 +34,10 @@ namespace IPC {
 class Decoder;
 class Encoder;
 
-namespace Detail {
-template<typename T, typename I> auto TestLegacyDecoder(int) -> decltype(I::decode(std::declval<Decoder&>(), std::declval<T&>()), std::true_type { });
-template<typename T, typename I> auto TestLegacyDecoder(long) -> std::false_type;
-template<typename T, typename I> auto TestModernDecoder(int) -> decltype(I::decode(std::declval<Decoder&>()), std::true_type { });
-template<typename T, typename I> auto TestModernDecoder(long) -> std::false_type;
-}
-template<typename T, typename I = T> struct HasLegacyDecoder : decltype(Detail::TestLegacyDecoder<T, I>(0)) { };
-template<typename T, typename I = T> inline constexpr bool HasLegacyDecoderV = HasLegacyDecoder<T, I>::value;
-template<typename T, typename I = T> struct HasModernDecoder : decltype(Detail::TestModernDecoder<T, I>(0)) { };
-template<typename T, typename I = T> inline constexpr bool HasModernDecoderV = HasModernDecoder<T, I>::value;
+template<typename T, typename I = T, typename = void> struct HasLegacyDecoder : std::false_type { };
+template<typename T, typename I> struct HasLegacyDecoder<T, I, std::void_t<decltype(I::decode(std::declval<Decoder&>(), std::declval<T&>()))>> : std::true_type { };
```

Comment:
I know why HasLegacyDecoder has two template arguments. But, the second argument isn't used. You can remove it.

Reply:
Both `T` and `I` are needed due to how `HasModernDecoder` and `HasLegacyDecoder` are used in WebKit, in IPC::Decoder. There, the `I` (as in Impl) is specified as the `ArgumentCoder` specialization, which might be custom-written.

Reply:
Oh sorry, I didn't understand that.